### PR TITLE
Bump milli

### LINKED ISSUE #1685: Sending an attribute with a malformed syntax in `sort` at search time does not raise an invalid_sort error
With sortable-attributes

```json
["title", "rates"]
```

If I send `sort` in this way at search time (missing :asc/:desc)
```json
{
    "sort": ["title"]
}
```
or if I send `sort` in this way (malformed :asc/:desc)
```json
{
    "sort": ["title:deesc"]
}
```

The API response is 
```json
{
    "message": "invalid criterion title:deesc",
    "errorCode": "internal",
    "errorType": "internal_error",
    "errorLink": "https://docs.meilisearch.com/errors#internal"
}
```

As a user, I would like to receive an `invalid_sort` error indicating a syntax error in the `sort` parameter.

### PR DESCRIPTION
**Pull request by irevoire**
fixes #1685, #1678, #1671, #1677 and #1680

### CODE REVIEW

File: meilisearch-http/src/error.rs [hunk 1]
```diff
@@ -93,6 +93,9 @@ impl ErrorCode for MilliError<'_> {
                     | UserError::InvalidDocumentId { .. }
                     | UserError::InvalidStoreFile
                     | UserError::NoSpaceLeftOnDevice
+                    | UserError::InvalidAscDescSyntax { .. }
+                    | UserError::InvalidSortName { .. }
+                    | UserError::SortRankingRuleMissing
```
Comment (Kerollmops) [inline meilisearch-http/src/error.rs:98]: Are you sure those errors must be considered internal?

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Could you please make sure that we correctly return [the expected error message](https://github.com/meilisearch/specifications/blob/develop/text/0055-sort.md), please?
> Sending a sort parameter while the sort ranking rule is not specified in the ranking rules settings will lead to a 400 Bad Request - invalid_sort error.

```json
{
    "message": "The sort ranking rule must be specified in the ranking rules settings to use the sort parameter at search time.",
    "errorCode": "invalid_sort",
    "errorType": "invalid_request_error",
    "errorLink": "https://docs.meilisearch.com/errors#invalid_sort"
}

```

**Review by Kerollmops — APPROVED** [review]
Thank you tamo!

**Review by Kerollmops — APPROVED** [review]
bors merge

**Review by curquiza — APPROVED** [review]

**Comment by curquiza** [comment]
bors merge-

**Review by curquiza — APPROVED** [review]
I changed the based branch from `main` to `release-v0.22.0`

bors merge

**Comment by curquiza** [comment]
bors merge

**Pull request merged**